### PR TITLE
Sync federated clouds using a cron job

### PR DESCRIPTION
**Pull request by DeepDiver1975**
Synchronization of user information between federated clouds is now done using a cron job once a day.

### CODE REVIEW

File: apps/federation/command/syncfederationaddressbooks.php [hunk 1]
```diff
@@ -42,30 +36,14 @@ protected function execute(InputInterface $input, OutputInterface $output) {
...
-			$targetBookProperties = [
-					'{DAV:}displayname' => $url
-			];
-			try {
-				$newToken = $this->syncService->syncRemoteAddressBook($url, 'system', $sharedSecret, $syncToken, $targetPrincipal, $targetBookId, $targetBookProperties);
-				if ($newToken !== $syncToken) {
-					$this->dbHandler->setServerStatus($url, TrustedServers::STATUS_OK, $newToken);
-				}
-			} catch (\Exception $ex) {
+		$this->syncService->syncThemAll(function($url, $ex) use ($progress, $output) {
```
Comment (MorrisJobke) [inline apps/federation/command/syncfederationaddressbooks.php]:
```
{"reqId":"g3rwKczOxhWAR2xlbk\/J","remoteAddr":"","app":"PHP","message":"Missing argument 2 for OCA\\Federation\\Command\\SyncFederationAddressBooks::OCA\\Federation\\Command\\{closure}(), called in \/Users\/morrisjobke\/Projects\/owncloud\/master\/apps\/federation\/lib\/syncfederationaddressbooks.php on line 33 and defined at \/Users\/morrisjobke\/Projects\/owncloud\/master\/apps\/federation\/command\/syncfederationaddressbooks.php#39","level":0,"time":"2016-01-25T09:44:44+00:00","method":"--","url":"--"}
{"reqId":"g3rwKczOxhWAR2xlbk\/J","remoteAddr":"","app":"PHP","message":"Undefined variable: ex at \/Users\/morrisjobke\/Projects\/owncloud\/master\/apps\/federation\/command\/syncfederationaddressbooks.php#40","level":0,"time":"2016-01-25T09:44:44+00:00","method":"--","url":"--"}
```

File: apps/federation/lib/syncfederationaddressbooks.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+		$this->syncService = \OC::$server->query('CardDAVSyncService');
+		$this->dbHandler = $dbHandler;
+	}
+
+	public function syncThemAll(\Closure $callback) {
+
+		$trustedServers = $this->dbHandler->getAllServer();
+		foreach ($trustedServers as $trustedServer) {
+			$url = $trustedServer['url'];
+			$callback($url);
```
Comment (MorrisJobke) [inline apps/federation/lib/syncfederationaddressbooks.php]:
```
{"reqId":"g3rwKczOxhWAR2xlbk\/J","remoteAddr":"","app":"PHP","message":"Missing argument 2 for OCA\\Federation\\Command\\SyncFederationAddressBooks::OCA\\Federation\\Command\\{closure}(), called in \/Users\/morrisjobke\/Projects\/owncloud\/master\/apps\/federation\/lib\/syncfederationaddressbooks.php on line 33 and defined at \/Users\/morrisjobke\/Projects\/owncloud\/master\/apps\/federation\/command\/syncfederationaddressbooks.php#39","level":0,"time":"2016-01-25T09:44:44+00:00","method":"--","url":"--"}
```

 (first line of the comment above)

**Comment by DeepDiver1975** [comment]
one should write tests .....

**Comment by DeepDiver1975** [comment]
THX

**Comment by DeepDiver1975** [comment]
@MorrisJobke issues have been addressed

**Comment by MorrisJobke** [comment]
The cronjob is triggered, but the users are not synced. If this should only address the cronjob then I'm fine with merging, but the sync of the users needs to be fixed in a followup PR.

**Comment by MorrisJobke** [comment]
:+1:

**Comment by DeepDiver1975** [comment]
> The cronjob is triggered, but the users are not synced.

please elaborate - sync of trusted servers was working weeks back

**Comment by MorrisJobke** [comment]
> please elaborate - sync of trusted servers was working weeks back

It doesn't show up in the remote server :see_no_evil: Let me test with master.

**Comment by DeepDiver1975** [comment]
> It doesn't show up in the remote server :see_no_evil: Let me test with master.

Who is *it*?
Furthermore this is a pull mechanism - not a push mechanism - so remote users have to show up in the local instance.

**Comment by MorrisJobke** [comment]
My local issue is covered by  https://github.com/owncloud/core/issues/21929 -> :+1: for this here.

**Comment by LukasReschke** [comment]
:+1:

**Pull request merged**